cmd/observer/node_builder/observer_builder.go

```diff
@@ -0,0 +1,552 @@
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/module/buffer"
+	finalizer "github.com/onflow/flow-go/module/finalizer/consensus"
+	"github.com/onflow/flow-go/module/id"
+	"github.com/onflow/flow-go/module/mempool/stdmap"
+	"github.com/onflow/flow-go/module/metrics"
+	"github.com/onflow/flow-go/module/synchronization"
+	"github.com/onflow/flow-go/network"
+	netcache "github.com/onflow/flow-go/network/cache"
+	cborcodec "github.com/onflow/flow-go/network/codec/cbor"
+	"github.com/onflow/flow-go/network/p2p"
+	"github.com/onflow/flow-go/network/validator"
+	"github.com/onflow/flow-go/state/protocol"
+	badgerState "github.com/onflow/flow-go/state/protocol/badger"
+	"github.com/onflow/flow-go/state/protocol/blocktimer"
+	storage "github.com/onflow/flow-go/storage/badger"
+)
+
+// ObserverBuilder extends cmd.NodeBuilder and declares additional functions needed to bootstrap an Access node
+// These functions are shared by staked and observer node builders.
+// The Staked network allows the staked nodes to communicate among themselves, while the public network allows the
+// observers and a staked Access node to communicate.
+//
+//                                 public network                           staked network
+//  +------------------------+
+//  | observer Node 1        |<--------------------------|
+//  +------------------------+                           v
+//  +------------------------+                         +--------------------+                 +------------------------+
+//  | observer Node 2        |<----------------------->| Staked Access Node |<--------------->| All other staked Nodes |
+//  +------------------------+                         +--------------------+                 +------------------------+
+//  +------------------------+                           ^
+//  | observer Node 3        |<--------------------------|
+//  +------------------------+
+
+type ObserverBuilder interface {
+	cmd.NodeBuilder
+
+	// IsStaked returns True if this is a staked Access Node, False otherwise
+	IsStaked() bool
+}
+
+// AccessNodeConfig defines all the user defined parameters required to bootstrap an access node
+// For a node running as a standalone process, the config fields will be populated from the command line params,
+// while for a node running as a library, the config fields are expected to be initialized by the caller.
+type AccessNodeConfig struct {
+	staked                       bool
+	bootstrapNodeAddresses       []string
+	bootstrapNodePublicKeys      []string
+	observerNetworkingKeyPath    string
+	bootstrapIdentities          flow.IdentityList // the identity list of bootstrap peers the node uses to discover other nodes
+	NetworkKey                   crypto.PrivateKey // the networking key passed in by the caller when being used as a library
+	supportsUnstakedFollower     bool              // True if this is a staked Access node which also supports observer nodes consensus follower engines
```

Comment:
observer doesn't need this

Reply:
Do you mean we do not need them in cmd/access?